app/actions/remote/team.ts

```diff
@@ -428,3 +497,50 @@ export async function handleKickFromTeam(serverUrl: string, teamId: string) {
         logDebug('Failed to kick user from team', error);
     }
 }
+
+export async function getTeamMembersByIds(serverUrl: string, teamId: string, userIds: string[], fetchOnly?: boolean) {
+    let client;
+
+    try {
+        client = NetworkManager.getClient(serverUrl);
+        const members = await client.getTeamMembersByIds(teamId, userIds);
+
+        if (!fetchOnly) {
+            setTeamLoading(serverUrl, true);
+
+            const teamMemberships: TeamMembership[] = [];
+            const roles: Record<string, boolean> = {};
+
+            for (const member of members) {
+                teamMemberships.push(member);
+                member.roles.split(' ').forEach((role) => {
+                    if (!roles[role]) {
+                        roles[role] = true;
+                    }
```

Comment:
For my understanding, why are we doing this?

Reply:
Just trying to get unique roles for fetching. Updated for a simpler approach using arrays.